### PR TITLE
hoc: use DCDO value hoc_year in many places

### PR DESCRIPTION
**Pull request by breville**
Rather than having to manually update a whole bunch of places with the year of the upcoming Hour of Code, instead use DCDO value `hoc_year` to specify that year.

This incorporates a bunch of suggestions given in https://github.com/code-dot-org/code-dot-org/pull/15735 to this end.

Note that it doesn't address the changes to www.hourofcode.com/events.  From perusing those files, it seems there is an opportunity for a more significant code refactoring but I want to consider it separate to this change.  Therefore [this](https://github.com/code-dot-org/code-dot-org/pull/15735/files#r121201116) suggestion is not yet addressed.

### CODE REVIEW

**Comment by breville** [comment]
@tanyaparker I've not yet embarked on testing these changes, and will probably need some help from you to cover everything.  But I wanted to give you and @ashercodeorg an early look at the approach.

File: lib/cdo/contact_rollups.rb [hunk 1]
```diff
@@ -75,17 +75,17 @@ class ContactRollups
...
-      {kind: "'HocSignup2017'", dest_field: "hoc_organizer_years", dest_value: "'2017'"},
-      {kind: "'Petition'", dest_field: "roles", dest_value: "'Petition Signer'"}
-    ].freeze
+  FORM_INFOS = []
+  FORM_INFOS << {kind: "'CSEdWeekEvent2013'", dest_field: "hoc_organizer_years", dest_value: "'2013'"}
+  [2014..hoc_year].each do |year|
+    FORM_INFOS << {kind: "'HocSignup#{year}'", dest_field: "hoc_organizer_years", dest_value: "'#{year}'"}
+  end
+  FORM_INFOS << {kind: "'Petition'", dest_field: "roles", dest_value: "'Petition Signer'"}
+  FORM_INFOS.freeze
```
Comment (breville) [inline lib/cdo/contact_rollups.rb:88]: I maintained order here, though not sure whether that was necessary.

**Comment by ashercodeorg** [comment]
Looks like we missed a file, namely `pegasus/routes/v2_form_routes`.

File: bin/cron/analyze_hoc_activity [hunk 2]
```diff
@@ -76,11 +76,13 @@ def main
     day += 1
   end
 
+  hoc_year = DCDO.get("hoc_year", 2017)
```
Comment (ashercodeorg) [inline bin/cron/analyze_hoc_activity:80]: This should necessitate a `require 'dynamic_config/dcdo'`?

Comment (breville) [reply bin/cron/analyze_hoc_activity:80]: Thanks.  Also found another similar case.  @tanyaparker and I are scheduled to test all of this early next week to make sure it's working as expected, before merging.

**Comment by tanyaparker** [comment]
@breville

**I didn't test these and need your help testing locally.**
1. Run bin/cron/analyze_hoc_activity
	1a. I got errors for this locally even on staging branch.
2. Run bin/cron/update_hoc_map
	2a. Make sure you see id # 6053412 in fusion table
	https://fusiontables.google.com/DataSource?docid=164RLDWEyPij3-Z9O0791ta_a8RfuCdGUkWRFcnBD#rows:id=1
3. Test that you see custom hoc scale messages.
	3a. Sign up on hoc.com with your email
	3b. Turn on scale mode
	3c. When logged out, try to create an account
	3d. You should see a custom message about the Hour of Code and not needing to create an account to participate.

**I didn't test these but can test on staging.**
1. Get correct "thank you email" from signing up a hoc event
2. See hoc events on /events page (could see production data on local)

**I tested this.**
1. Can sign up for hoc event

**Comment by ashercodeorg** [comment]
Re: (1a), I'm working on a fix for this.

@tanyaparker, my assumption is that you are seeing the same `only_full_group_by` errors that I'm seeing?

**Comment by tanyaparker** [comment]
@ashercodeorg I actually got a nil error on line 137.

`existing_lines_of_code = Properties.get(:metrics)['lines_of_code']`

**Comment by ashercodeorg** [comment]
Ah, that is much, much simpler to fix. From `pegasus-console` run:
1. `require 'cdo/properties'`
2. `Properties.set :metrics, {'lines_of_code' => 123}`
Though it might make sense to guard against the key missing (both `:metrics` and `'lines_of_code'`), I'm not certain that change makes sense in all environments. This'll fix it for you locally.

**Comment by tanyaparker** [comment]
That fixed it thank you!

**Comment by breville** [comment]
Regarding those three scenarios to test:

1. I'm told that was fixed.
2. Testing that separate with the PR to update the API.
3. Even in scale mode I was able to sign up an event on hourofcode.com and a user on code.org.  There is a custom message shown when trying to sign up with an email already used for an event, but that seems independent of scale mode.

Merging now.

**Comment by tanyaparker** [comment]
Testing in staging... Unfortunately the Spanish/Portuguese emails don't work, you always get English. But this just keeps the existing behavior (it never worked). So I'll track to fix when we do more hoc work after back to school.

**Comment by ashercodeorg** [comment]
Do we have an understanding why the i18n emails aren't sent? If we do, I imagine it wouldn't be too hard to fix.

**Comment by tanyaparker** [comment]
My guess is that we don't know `@country` at the time that we're choosing which email template to send. I know we must know it when someone signs up for hoc because we track sign ups per country page. But not sure if it passes all the way to the email template.

**Comment by ashercodeorg** [comment]
Looks like you are right, I'm not seeing the `@country` variable getting set anywhere. That said, it should be easy enough to set.

**Pull request merged**